File: 242.valid-anagram.go
/*
 * @lc app=leetcode id=242 lang=golang
 *
 * [242] Valid Anagram
 */

// @lc code=start
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	m := make(map[rune]int)
	n := make(map[rune]int)
	
	for i, u := range s {
		m[u]++
		n[rune(t[i])]++
	}
	
	for k, v := range m {
		if v != n[k] {
			return false
		}
	}
	
	return true
}
// @lc code=end

File: 242.valid-anagram.rs
/*
 * @lc app=leetcode id=242 lang=rust
 *
 * [242] Valid Anagram
 */

// @lc code=start
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let mut m = std::collections::HashMap::new();
        s.chars().for_each(|c| *m.entry(c).or_insert(0) += 1);
        t.chars().for_each(|c| *m.entry(c).or_insert(0) -= 1);
        m.into_values().all(|v| v == 0)
    }
}
// @lc code=end

